### PR TITLE
Implemented state change

### LINKED ISSUE #3: state doesn't change from home to not_home
Environment:
Sagemcom F@st 3865bProximus (b-box3)
Release: 8c.514A
GUI Version
7.4.2.0

The state of my tracked device doesn;t change from 'home' to 'not_home'. if i turn off the wifi it keeps the home state then if i reload the integration the Mobile device entity it shows up as 'This entity is not currently available.'

I need to spend some time to investigate why i cannot get the logs and i will have a look into the code. Any other ideas how can debug your application? HA Core is running over docker in Synology For your reference here we have the my config file. 

```
default_config:
frontend:
  themes: !include_dir_merge_named themes

tts:
  - platform: google_translate

group: !include groups.yaml
automation: !include automations.yaml
script: !include scripts.yaml
logger:
  default: critical
  logs:
    custom_components.sagemcom_fast: debug
```

### PR DESCRIPTION
**Pull request by arunpoudel**
Implemented state change which resolves #3

### CODE REVIEW

**Comment by iMicknl** [comment]
Great work @arunpoudel and thanks for your contribution. I will have a look in the coming days, but a small remark would be that the scan interval would be better in an OptionsFlow. Have you ever written this? Otherwise I am happy to tackle this in a follow up PR.

The linting is failing because you need a `version` key in the manifest, but I am happy to tackle this one in a separate PR as well.

**Comment by arunpoudel** [comment]
I have not worked with OptionsFlow, just started installing home assistance at my place, and saw the issue open. I can have a look at it over the weekend and update the pull request.

**Comment by iMicknl** [comment]
Cool! And in case you need help, let me know :).

Happy with your contribution, it was on my todolist for a while, but I couldn't get to it. They will be happy on the [Home Assistant Community](https://community.home-assistant.io/t/sagemcom-router-support/179880/47).

**Comment by arunpoudel** [comment]
Have implemented OptionsFlow. But the option is not reflected until homeassistance is restarted.

**Comment by iMicknl** [comment]
> Have implemented OptionsFlow. But the option is not reflected until homeassistance is restarted.

That was quick! In order to achieve this, you would need an [update listener](https://developers.home-assistant.io/docs/config_entries_options_flow_handler/). 

You can find a end to end example here: https://github.com/iMicknl/ha-tahoma/blob/master/custom_components/tahoma/__init__.py

File: custom_components/sagemcom_fast/device_tracker.py [hunk 1]
```diff
@@ -15,42 +25,80 @@
...
-class SagemcomScannerEntity(ScannerEntity, RestoreEntity):
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+
+    async_add_entities(
+        SagemcomScannerEntity(coordinator, idx, config_entry.entry_id)
+        for idx, device in coordinator.data.items()
+    )
+
+
+class SagecomDataUpdateCoordinator(DataUpdateCoordinator):
```
Comment (iMicknl) [inline custom_components/sagemcom_fast/device_tracker.py:36]:
```suggestion
class SagemcomDataUpdateCoordinator(DataUpdateCoordinator):
```

File: custom_components/sagemcom_fast/options_flow.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+"""Options flow for Sagemcom integration."""
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_SCAN_INTERVAL
+import voluptuous as vol
+
+SCAN_INTERVAL = 10
```
Comment (iMicknl) [inline custom_components/sagemcom_fast/options_flow.py:7]:
```suggestion
DEFAULT_SCAN_INTERVAL = 10
```
Could you rename it and move it to `const.py`?

File: custom_components/sagemcom_fast/options_flow.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class OptionsFlow(config_entries.OptionsFlow):
+    """Handle a options flow for Sagemcom."""
+
+    def __init__(self, config_entry):
+        """Initialize Sagecom options flow."""
```
Comment (iMicknl) [inline custom_components/sagemcom_fast/options_flow.py:14]:
```suggestion
        """Initialize Sagemcom options flow."""
```

File: custom_components/sagemcom_fast/options_flow.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        self._config_entry = config_entry
+        self.options = dict(config_entry.options)
```
Comment (iMicknl) [inline custom_components/sagemcom_fast/options_flow.py:16]: Not sure if this one is required.

File: custom_components/sagemcom_fast/options_flow.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL,
+                        default=self._config_entry.options.get(
+                            CONF_SCAN_INTERVAL, SCAN_INTERVAL
+                        ),
+                    ): int
```
Comment (iMicknl) [inline custom_components/sagemcom_fast/options_flow.py:33]:
```suggestion
                    ): cv.positive_int
```

or

```suggestion
                    ): vol.All(cv.positive_int, vol.Clamp(min=MIN_UPDATE_INTERVAL))
```

Best would be to make sure that users don't accidentally set it to negative numbers.

**Review by iMicknl — COMMENTED** [review]
Great work @arunpoudel! I had some minor remarks, mainly around the naming. 

If you have time, it would be great if you could also add the information to strings.json ([example](https://github.com/iMicknl/ha-tahoma/blob/master/custom_components/tahoma/strings.json) and to en.json.

**Comment by iMicknl** [comment]
@arunpoudel let me know if you are able to make the required changes, otherwise I could do it and push to this PR as well :). Appreciate you taking the time to work on this.

Which Sagemcom F@st device do you have?

**Review by iMicknl — APPROVED** [review]
Great work! Thanks for your contribution.

**Comment by iMicknl** [comment]
Oops, maybe I merged it too soon. It seems that the integration is broken / the config flow doesn't work.

I will have access tomorrow to my installation where the Sagemcom router is running, so I will have a look than. Unfortunately my VPN connection went down..

**Comment by arunpoudel** [comment]
Ah, yes. The integration is still broken. I am still making changes to it. Can you reopen the PR, and revert the merge please?

**Comment by arunpoudel** [comment]
Will send in a new PR. My router is F@st 3890V3.

**Comment by iMicknl** [comment]
> Ah, yes. The integration is still broken. I am still making changes to it. Can you reopen the PR, and revert the merge, please?

My apologies, apparently the build pipeline for the underlying API wasn't working, thus there was an issue with me bumping to the latest version. That should now be resolved and this version should now be on PyPi.

**Pull request merged**